stats/src/main/java/io/airlift/stats/cardinality/DenseHll.java

```diff
@@ -249,16 +249,14 @@ public void insert(int bucket, int value)
         }
 
         if (delta > MAX_DELTA) {
-            int overflow = delta - MAX_DELTA;
+            byte overflow = (byte) (delta - MAX_DELTA);
 
-            if (!setOverflow(bucket, overflow)) {
-                // grow overflows arrays if necessary
-                overflowBuckets = Ints.ensureCapacity(overflowBuckets, overflows + 1, OVERFLOW_GROW_INCREMENT);
-                overflowValues = Bytes.ensureCapacity(overflowValues, overflows + 1, OVERFLOW_GROW_INCREMENT);
-
-                overflowBuckets[overflows] = bucket;
-                overflowValues[overflows] = (byte) overflow;
-                overflows++;
+            int overflowEntry = findOverflowEntry(bucket);
+            if (overflowEntry != -1) {
```

Comment:
Why this makes it faster? It introduces branching

Reply:
`delta > MAX_DELTA` is statistically unlikely, so this branch doesn't hit often. In any case, the old code already had a branch `if (!setOverflow(...))`. This just moves things around a bit.